dotnet/src/InternalUtilities/src/Diagnostics/ModelDiagnostics.cs

```diff
@@ -0,0 +1,246 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace Microsoft.SemanticKernel;
+
+[ExcludeFromCodeCoverage]
+internal static class ModelDiagnostics
+{
+    private static readonly string s_namespace = typeof(ModelDiagnostics).Namespace!;
+    private static readonly ActivitySource s_activitySource = new(s_namespace);
+
+    private const string EnableModelDiagnosticsSettingName = "Microsoft.SemanticKernel.Experimental.EnableModelDiagnostics";
+    private const string EnableSensitiveEventsSettingName = "Microsoft.SemanticKernel.Experimental.EnableModelDiagnosticsWithSensitiveData";
+
+    private static readonly bool s_enableModelDiagnostics = AppContextSwitchHelper.GetConfigValue(EnableModelDiagnosticsSettingName);
+    private static readonly bool s_enableSensitiveEvents = AppContextSwitchHelper.GetConfigValue(EnableSensitiveEventsSettingName);
```

Comment:
```suggestion
    private static readonly bool s_enableSensitiveEvents = AppContextSwitchHelper.GetConfigValue(EnableSensitiveEventsSettingName);
    private static readonly bool s_enableModelDiagnostics = s_enableSensitiveEvents 
 || AppContextSwitchHelper.GetConfigValue(EnableModelDiagnosticsSettingName);
```

Reply:
This was actually my original implementation. And then I thought "what would happen if the someone accidentally change the order of the two statements?". It won't generate any error, but the behavior will be very different.